gcloud-java-dns/src/main/java/com/google/gcloud/dns/spi/DefaultDnsRpc.java

```diff
@@ -103,7 +103,7 @@ public boolean deleteZone(String zoneName) throws DnsException {
       dns.managedZones().delete(this.options.projectId(), zoneName).execute();
       return true;
     } catch (IOException ex) {
-      DnsException serviceException = translate(ex);
+      DnsException serviceException = translate(ex, false);
```

Comment:
I do not think so. It would be the same case as create. If a zone exists, it will be deleted with success (200) return value. If the zone does not exist, the service returns 404. So, if the socket times out but the delete is processed, the retry will receive in a subsequent call 404 and return `false` instead of `true`.